### PR TITLE
jewel: Upgrading 0.94.6 -> 0.94.9 saturating mon node networking

### PR DESCRIPTION
**Pull request by dachary**
http://tracker.ceph.com/issues/17734
http://tracker.ceph.com/issues/17694

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -3061,6 +3061,25 @@ void OSDMonitor::get_health(list<pair<health_status_t,string> >& summary,
       }
     }
 
+    // warn about upgrade flags that can be set but are not.
+    if ((osdmap.get_up_osd_features() & CEPH_FEATURE_SERVER_KRAKEN) &&
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: should remove the code referencing KRAKEN.

**Comment by dachary** [comment]
@tchaikov removed reference to kraken in **mon/OSDMonitor: health warn if require_{jewel,kraken} flags aren't set ** and **mon/OSDMonitor: encode canonical full osdmap based on osdmap flags**

**Comment by dachary** [comment]
<pre>
teuthology-suite -k distro --verbose --suite upgrade/jewel-x --suite-branch jewel --ceph wip-17734-jewel --machine-type vps --priority 101 machine_types/vps.yaml
</pre>

* **running** http://pulpito.ceph.com/loic-2016-10-28_18:18:59-upgrade:jewel-x-wip-17734-jewel-distro-basic-vps/

<pre>
teuthology-suite -k distro --priority 101 --suite rados --subset $(expr $RANDOM % 2000)/2000 --suite-branch master --email loic@dachary.org --ceph wip-17734-jewel --machine-type smithi
</pre>

* **running** http://pulpito.ceph.com/loic-2016-10-28_18:14:04-rados-wip-17734-jewel-distro-basic-smithi/

<pre>
teuthology-suite -k distro --verbose --suite upgrade/hammer-x --suite-branch jewel --ceph wip-17734-jewel --machine-type vps --priority 101 machine_types/vps.yaml
</pre>

* **running** http://pulpito.ceph.com/loic-2016-10-28_18:22:11-upgrade:hammer-x-wip-17734-jewel-distro-basic-vps/

**Comment by dachary** [comment]
I'm concerned by http://pulpito.ceph.com/loic-2016-10-31_09:24:48-rados-wip-17734-jewel-distro-basic-smithi/505937/ which looks like another form of http://tracker.ceph.com/issues/17728#note-6 . The rest of the failures above are either fixed or known.

**Comment by dachary** [comment]
http://pulpito.ceph.com/loic-2016-11-07_06:19:54-rados-wip-17734-jewel-distro-basic-smithi/ has a clean run. It still is racy but no more than master and that shows the commits in this pull request do the right thing.

**Comment by dachary** [comment]
@tchaikov although some upgrade tests are still missing because VPS run have had issues since last thursday, I think there is enough proof that these commits do the right thing and that we're in no danger of a regression. What say you ?

**Comment by tchaikov** [comment]
@dachary cool, let's merge it!

**Comment by smithfarm** [comment]
It appears that, as of 10.2.4, cluster admins will have to do "ceph osd set require_jewel_osds", otherwise the MONs will complain: "all OSDs are running jewel or later but the 'require_jewel_osds' osdmap flag is not set". Does this deserve a mention in the 10.2.4 release notes?

**Comment by smithfarm** [comment]
(answering my own question based on clarification provided by @liewegas and @athanatos on IRC)

When the last hammer OSD in a cluster containing jewel MONs is upgraded to jewel, as of 10.2.4 the jewel MONs will issue this warning: "all OSDs are running jewel or later but the 'require_jewel_osds' osdmap flag is not set" and change the cluster health status to HEALTH_WARN.

This is a signal for the admin to do "ceph osd set require_jewel_osds" - by doing this, the admin acknowledges that there is no downgrade path.

(I propose that we add this text, or one like it, to the 10.2.4 release notes.)

**Comment by liewegas** [comment]
+1

**Pull request merged**